docs/source/guide/zne-1-intro.myst

```diff
@@ -0,0 +1,137 @@
+
+```{code-cell} ipython3
+from mitiq import benchmarks
+
+circuit = benchmarks.generate_rb_circuits(
+  n_qubits=1, num_cliffords=2, return_type = frontend,
+)[0]
+
+print(circuit)
+```
+
+## Define the executor function
+We refer to a user-defined function that inputs a quantum program and executes it on a 
+backend as an executor. Such functions always accept a quantum program, sometimes 
+accept other arguments, and always return an expectation value as a float. As an 
+example, below we define a simple executor function which inputs a Cirq circuit, 
+executes it on a noisy simulator, and returns the probability of the ground state. 
+See the [Executors](guide-executors.myst) section for more information on how to 
+define more advanced executors.
+
+```{code-cell} ipython3
+import numpy as np
+from cirq import DensityMatrixSimulator, depolarize
+from mitiq.interface import convert_to_mitiq
+
+def executor(circuit, noise_level=0.01):
+    """Returns Tr[ρ |0⟩⟨0|] where ρ is the state prepared by the circuit
+    executed with depolarizing noise.
+    """
+    # Replace with code based on your frontend and backend.
+    mitiq_circuit, _ = convert_to_mitiq(circuit)
+    noisy_circuit = mitiq_circuit.with_noise(depolarize(p=noise_level))
+    rho = DensityMatrixSimulator().simulate(noisy_circuit).final_density_matrix
+    return rho[0, 0].real
+```
+
+The [`executor`](https://mitiq.readthedocs.io/en/latest/guide/guide-executors.html?highlight=mitiq.executor#executors) can be used to evaluate noisy (unmitigated) expectation values.
+
+```{code-cell} ipython3
+# Compute the expectation value of the |0><0| observable.
+noisy_value = executor(circuit)
+ideal_value = executor(circuit, noise_level=0.0)      
+print(f"Error without mitigation: {abs(ideal_value - noisy_value) :.5f}")
+```
+
+## Select a noise scaling method
+In Mitiq, folding functions input a circuit and a scale factor, i.e., 
+a floating point value which corresponds to (approximately) how much the length 
+of the circuit is scaled. The minimum scale factor is one (which corresponds to 
+folding no gates). A scale factor of three corresponds to folding all gates locally. 
+Scale factors beyond three begin to fold gates more than once.
+
++++
+
+For local folding, there is a degree of freedom for which gates to fold first. 
+The order in which gates are folded can have an important effect on how the noise is 
+scaled. The function [`mitiq.zne.scaling.folding.fold_gates_at_random()`](https://mitiq.readthedocs.io/en/latest/apidoc.html#mitiq.zne.scaling.folding.fold_gates_at_random)
+returns a new folded circuit by applying the map G -> G G^dag G to a subset of gates 
+of the input circuit, starting with gates at the right (end) of the circuit.
+
+The folded circuit has a number of gates approximately equal to scale_factor * n where 
+n is the number of gates in the input circuit. See the section on [noise scaling by 
+unitary folding](zne-5-theory.myst) for more details.
```

Comment:
I would not give too many details on folding functions at this stage. Note also that *noise scaling functions* can be more general than folding functions. 
I would propose the following rewording of the text but feel free to change it.

Note: if the suggested {func}`.fold_gates_at_random()` doesn't work, you can try with {func}`~mitiq.zne.scaling.folding.fold_gates_at_random()`. If this also doesn't work, I am fine with adding a manual link as you did but replacing `latest` with `stable` in the url: `https://mitiq.readthedocs.io/en/stable/apidoc.html#mitiq.zne.scaling.folding.fold_gates_at_random`.

```suggestion
In Mitiq, one can select a noise scaling method via *noise scaling functions*.
A noise scaling function takes as input a circuit and a real scale factor and
returns a new circuit. The returned circuit is equivalent to the input one (if executed on a noiseless backend),
but is more sensitive to noise when executed on a real noisy backend. In practice, by applying a noise
scaling function before the execution of a circuit, one can indirectly scale up the effect of noise. 
See [What additional options are available in ZNE?](zne-3-options.myst) for more details.

For example, the function {func}`.fold_gates_at_random()` applies the *unitary folding* map $G -> G G^dag G$
to a random subset of gates of the input circuit.
The folded circuit is more sensitive to gate errors since it has a number of gates approximately
equal to `scale_factor * n`, where `n` is the number of gates in the input circuit. 
```